test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java

```
@@ -99,6 +100,20 @@
                 File.separator + "test-classes",
                 File.separator + "classes");
         //endregion
+
+        String mappings = System.getenv("TEST_TO_MAIN_MAPPINGS");
+        if (mappings != null) {
+            Stream.of(mappings.split(","))
+                    .filter(s -> !s.isEmpty())
+                    .forEach(s -> {
+                        String[] entry = s.split(":");
+                        if (entry.length == 2) {
+                            TEST_TO_MAIN_DIR_FRAGMENTS.put(entry[0], entry[1]);
+                        } else {
+                            System.err.println("Unable to parse additional test-to-main mapping: " + s);
```

I think we should throw an exception in this case instead.